app/controllers/homepage_controller.rb

```diff
@@ -154,26 +157,30 @@ def find_listings(params, listings_per_page, filter_params, includes, location_s
     dropdowns = filter_params[:custom_dropdown_field_options].map { |dropdown_field| dropdown_field.merge(type: :selection_group, operator: :or) }
     numbers = numeric_search_params.map { |numeric| numeric.merge(type: :numeric_range) }
 
-    coordinates = Maybe(params[:lc]).map { search_coordinates(params[:lc]) }.or_else({})
     distance_unit = (location_search_in_use && MarketplaceService::API::Api.configurations.get(community_id: @current_community.id).data[:distance_unit] == :metric) ? :km : :miles
+    location_search_hash = location_search_params(
+      params[:lc],
+      distance_unit,
+      Maybe(params[:distance_max])
+        .map { |d| [APP_MINIMUM_DISTANCE_MAX, d.to_f].max }
+        .or_else(nil)
+    )
 
     search = {
       # Add listing_id
       categories: filter_params[:categories],
       listing_shape_ids: Array(filter_params[:listing_shape]),
       price_cents: filter_params[:price_cents],
       keywords: filter_params[:search],
-      latitude: coordinates[:latitude],
-      longitude: coordinates[:longitude],
-      distance_unit: distance_unit,
       fields: checkboxes.concat(dropdowns).concat(numbers),
       per_page: listings_per_page,
       page: Maybe(params)[:page].to_i.map { |n| n > 0 ? n : 1 }.or_else(1),
       price_min: params[:price_min],
       price_max: params[:price_max],
+      sort: nil,
```

Comment:
I think it would be more clear if the `sort` param was decided here, instead of in `location_search_params`